client/app/lib/getQueryResultData.js

```diff
@@ -14,3 +15,7 @@ export default function getQueryResultData(queryResult) {
     metadata: get(queryResult, "query_result.data.metadata", {}),
   };
 }
+
+export function useQueryResultData(queryResult) {
+  return useMemo(() => getQueryResultData(queryResult), [queryResult]);
```

Comment:
Shouldn't it depend on `queryResult.status` as well? I'll try to explain my concern. We get a new `queryResult` object each time we refresh query. But while query job is running - `queryResult` remains the same and just updates its fields (please correct me if I miss something). So if we'll pass a new `queryResult` somewhere - this hook will cache its data when it's "waiting"/"processing"/etc. and when query result will reach "done" state - `useMemo` will still return the same cached object.

Reply:
I see your concern now. Well, while this shouldn't be an issue as it seem that every place we use that hook we only take "done" queryResults, I also see the potential problem. Will add the dep 👍